lib/rubocop/cop/github/render_literal_helpers.rb

```diff
@@ -32,6 +28,10 @@ module RenderLiteralHelpers
           (send nil? {:render :render_to_string} (send _ :new ...) ...)
         PATTERN
 
+        def_node_matcher :render_view_component_instance_with_content?, <<-PATTERN
+          (send nil? {:render :render_to_string} (send (send _ :new ...) `:with_content ...))
```

Comment:
I think we could make this even more generic (if simpler). If anything has a `new` or a constant I'd be happy to consider it a view component and ignore it.